### ISSUE TITLE
Controller --service-sync-period is not respected

### ISSUE THREAD

**Issue by khenidak** [kubernetes/kubernetes#58776]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
The service sync loop is too aggressive and does not respect ```--service-sync-period``` as documented here https://kubernetes.io/docs/reference/generated/kube-controller-manager/ nor does it default to 5 mins. Calls are spawn off immediately to cloud provider's ```EnsureLoadBalancer()``` 

> A quick check against https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/service/service_controller.go#L205 shows that the flag is not used at all.

**What you expected to happen**:
Service Controller respecting the ```--service-sync-period``` period argument 

**How to reproduce it (as minimally and precisely as possible)**:

1- Create a cluster on any environment (This test was done on Azure, but wouldn't be different for any cloud provider).
2- Create Service (type=LoadBalancer) via
```
kubectl run n01 --image nginx --replicas 3 
kubectl expose deployment n01 --type=LoadBalancer --port 80
```
3- Watch service creation on load balancer via ```watch kubectl get svc``` or just watch controller logs as it calls cloud provider's ```EnsureLoadBalancer()```
4- The behavior is the same when deleting a service

The test ran with a controller with ```--service-sync-period=6m``` argument
 
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.7.9
- Cloud provider or hardware configuration: Azure (And others)
- OS (e.g. from /etc/os-release): Ubuntu 16.04.3 LTS (Xenial Xerus)
- Kernel (e.g. `uname -a`): Linux k8s-master-30081023-0 4.13.0-1005-azure #7-Ubuntu SMP Mon Jan 8 21:37:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: acs-engine
- Others:

@kubernetes/sig-network-bugs 
@kubernetes/sig-azure-misc 
@jdumars

**Comment by deads2k** [comment]
Do you have a reference for the last time this worked?  A brief check looks to me like it has been broken since at least v1.4 and no one has noticed.  YAGNI seems to apply.

**Comment by khenidak** [comment]
@deads2k ok, can I remove the flag from controller options? just to remove confusion

**Comment by deads2k** [comment]
> @deads2k ok, can I remove the flag from controller options? just to remove confusion

I think so.  Open the pull, I'll leave a few days for comments before merging.

**Comment by khenidak** [comment]
closed with #59359
